shell/common/isolate_configuration.h

```diff
@@ -19,30 +19,139 @@
 
 namespace flutter {
 
-/// Abstract Base Class that will configure a |DartIsolate|.
+//------------------------------------------------------------------------------
+/// @brief      An isolate configration is a collection of snapshots and asset
+///             managers that the engine will use to configure the isolate
+///             before invoking its root entrypoint. The set of snapshots must
+///             be sufficient for the engine to move the isolate from the
+///             |DartIsolate::Phase::LibrariesSetup| phase to the
+///             |DartIsolate::Phase::Ready| phase. Note that the isolate
+///             configuration will not be collected till the isolate tied to the
+///             configuration as well as any and all child isolates of that
+///             isolate are collected. The engine may ask the configuration to
+///             prepare multiple isolates. All subclasses of this class must be
+///             thread safe as the configuration may be created, collected and
+///             used on multiple thread. Usually these threads are engine or VM
+///             managed so care must be taken to ensure that subclasses do not
+///             reference any thred local state.
+///
 class IsolateConfiguration {
  public:
+  //----------------------------------------------------------------------------
+  /// @brief      Attempts to infer the isolate configuration from the settings
+  ///             object. If the VM is configured for AOT mode, snapshot
+  ///             resolution is attempted with predefined symbols present in the
+  ///             currently loaded process. In JIT mode, Dart kernel file
+  ///             resolution is attempted in the assets directory. If an IO
+  ///             worker is specified, snapshot resolution may be attempted on
+  ///             the serial worker task runner. The worker task runner thread
+  ///             must remain valid and running till after the shell associated
+  ///             with the engine used to launch the isolate for which this run
+  ///             configuration is used is collected.
+  ///
+  /// @param[in]  settings       The settings
+  /// @param[in]  asset_manager  The asset manager
+  /// @param[in]  io_worker      An optional IO worker. Sepcify `nullptr` is a
+  ///                            worker should not be used or one is not
+  ///                            available.
+  ///
+  /// @return     An isolate configuration if one can be inferred from the
+  ///             settings. If not, returns `nullptr`.
+  ///
   static std::unique_ptr<IsolateConfiguration> InferFromSettings(
       const Settings& settings,
       std::shared_ptr<AssetManager> asset_manager,
       fml::RefPtr<fml::TaskRunner> io_worker);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Create an AOT isolate configuration using snapshot symbols
```

Comment:
Here and below:
nit: Creates (use a third-person verb for consistency with above.

Reply:
Done.